### DISCUSSION TITLE
Membaca Data Excel dg Python & R

### DISCUSSION THREAD

**Discussion by joejnt** [teamalgoritma/community#866] [Q&A] [answered]
Dear Team Algoritma,

Jika saya mempunya 8 file excel yang namanya berbeda-beda, kemudian setiap file excel memiliki sheet yang namanya berbeda namun isi didalam sheet strukturnya sama. Bagaimana caranya membaca file excel tersebut tanpa menyimpannya dalam 1 object terlebih dahulu dengan menggunakan python dan R. Untuk saat ini saya menggunakan R dengan syntax berikut:

# Set path ke folder tempat file Excel disimpan
path_to_files <- "Data Input/28/28/"

# Mendapatkan daftar semua file Excel di folder tersebut
excel_files <- list.files(path_to_files, pattern = "\\.xlsx$", full.names = TRUE)

read_excel_sheets <- function(file_path) {
  sheet_names <- excel_sheets(file_path)
  sheets_data <- map_df(sheet_names, ~ read_excel(file_path, sheet = .x))
  return(sheets_data)
}

all_data <- map_df(excel_files, read_excel_sheets)
saveRDS(all_data, "Data Input/Raw Data Yesterday.rds")

Masalah yang saya hadapi adalah untuk menjalankan syntax ini membutuhkan waktu yang cukup lama, kurang lebih 20 menit karena total obs sampai dengan 5 juta (jika file sudah digabungkan).

Saya juga masih belum mendapatkan cara yang sama jika dijalankan menggunakan bahasa pemrograman python. Mohon saran dari rekan-rekan.

Best regards,
JT

**Comment by ramdhinafinita** [comment, marked as answer]
Dear @joejnt 
Jika yang diinginkan adalah menggabungkan beberapada data excel dimana didalamnya ada sheet_name yang akan digabungkan juga bisa coba untuk melakukan for loop seperti berikut ini dengan cara menggunakan python library Pandas. Kurang lebih caranya sebagai berikut:

![image](https://github.com/teamalgoritma/community/assets/64061969/198a8b58-2e8b-4f97-a9ee-4f1c38c655f6)

Jika jawaban ini membantu, silahkan untuk mark as answer pada jawaban tersebut. Thanks and happy learning!!

**Reply by joejnt** [reply]
dear @ramdhinafinita 

Terima kasih untuk informasinya.

Best regards,
JT
